Sources/PitchSpeller/Data Structures/_Graph.swift

```diff
@@ -82,7 +82,14 @@ struct _Graph<Weight: Weightedness, Pair: SymmetricPair & Directedness & Hashabl
     }
     
     func neighbors (of source: Node) -> [Node] {
-        
+        return nodes.compactMap{
```

Comment:
Can this just be:

```Swift
return nodes.compactMap { adjacents[Pair(source,$0)] }
```

Reply:
Also, should neighbors be ordered (`Array`), or would a `Set` make more sense?

Reply:
(Because it looks like your testing to ensure that order doesn't matter, so why not encode that in the type?)

Reply:
Oh whoops, I misread the code a little bit. I'm gonna pull down and toy with that.

Reply:
This works for me:

```Swift
return nodes.compactMap { adjacents.keys.contains(Pair(source,$0)) ? $0 : nil }
```

Reply:
To be slightly more clear:

```Swift
return nodes.compactMap { destination in
    adjacents.keys.contains(Pair(source,destination)) ? destination : nil
}
```